frontend/src/MainContainer.jsx

```diff
@@ -0,0 +1,14 @@
+import NavBar from "./components/NavBar";
+import { useNavigation } from "react-router-dom";
+import LoadingAnimation from "./components/LoadingAnimation";
+
+export const MainContainer = ({ children }) => {
+  const { state } = useNavigation();
+
+  return (
+    <div>
+      <NavBar />
```

Comment:
It's nice to move the `NavBar` to the top-level component, but I think here bowtie_version will not be rendered. As we have created a context provider for the version, we can utilize it here.

Reply:
What do you mean? I access version context in `NavBar` component

Reply:
Oops! Sorry, I overlooked it. :)